activestorage/s3.py

```diff
@@ -42,16 +42,24 @@ def reduce_chunk(server, username, password, source, bucket, object, offset,
     if filters is not None:
         raise NotImplementedError("Filters are not yet supported!")
 
-    request_data = build_request_data(source, bucket, object, offset, size, compression, filters, missing, dtype, shape, order, chunk_selection)
+    request_data = build_request_data(source, bucket, object, offset, size,
+                                      compression, filters, missing, dtype,
+                                      shape, order, chunk_selection)
     api_operation = "sum" if operation == "mean" else operation or "select"
     url = f'{server}/v1/{api_operation}/'
-    response = request(url, username, password, request_data)
+    try:
+        response = request(url, username, password, request_data)
+    except requests.exceptions.ConnectionError as exc:
```

Comment:
There are various exceptions that requests might raise. You could try to handle them all, or catch `RequestException`.

https://requests.readthedocs.io/en/latest/_modules/requests/exceptions/